src/media/Icon.jsx

```diff
@@ -14,6 +37,53 @@ export const ICON_CLASS = 'svg';
  * @module Icon
  */
 class Icon extends React.PureComponent {
+	constructor(props){
+		super(props);
+		this.handleMediaChange = this.handleMediaChange.bind(this);
+
+		this.state = {
+			iconScaleFactor: 1
+		};
+	}
+
+	handleMediaChange() {
+		if (!this.state.mediaQueries) {
+			return;
+		}
+
+		this.setState((state) => ({
+			iconScaleFactor: getScaleFactor({
+				isMedium: state.mediaQueries.medium.matches,
+				isLarge: state.mediaQueries.large.matches
+			})
+		}));
+	}
+
+	componentDidMount() {
```

Comment:
We should have some tests here around what componentWillMount is doing along with the above handleMediaChange + comments around what these functions are doing